clover_api.rb

```diff
@@ -34,6 +34,42 @@ class CloverApi < Roda
     only_json? true
     use_jwt? true
 
+    # Converting rodauth error response to the common error format of the API
+    json_response_body do |hash|
+      # In case of an error, rodauth returns the error in the following format
+      # {
+      #   (required) "error": "There was an error logging in"
+      #   (optional) "field-error": [
+      #     "password",
+      #     "invalid password"
+      #   ]
+      # }
+      if hash["error"]
+        error_message = hash["error"]
+        type, code = case error_message
+        when "There was an error logging in"
+          ["Invalid Credentials", 401]
+        when "invalid JWT format or claim in Authorization header"
+          ["Invalid Request", 400]
+        when "Please login to continue"
+          ["Login Required", 401]
+        else
+          # :nocov:
+          ["Authentication Error", 401]
+          # :nocov:
+        end
+
+        hash.clear
+        hash["error"] = {
+          "code" => code,
+          "type" => type,
+          "message" => error_message
+        }
```

Comment:
Why do we clear hash instead of return a new one?
```suggestion
        return {
            "error" => {
                "code" => code,
                "type" => type,
                "message" => error_message
            }
        }.to_json
```

Reply:
I wanted to explicitly show that we are overwriting the hash returned by rodauth, let me keep it as it is.